### PR TITLE
Added a few tweaks to enable table customization and styles for paragraphs

### PR DESCRIPTION
**Pull request by dcharles525**
Hello! Below are list of changes made:

1. Added the ability to customize table width from the AddTable func
2. Added the ability to customize the border colors from the AddTable func
3. Added a function under the Paragraph struct to accept style names
4. Added tables tests modifications to make the tests run

Let me know what I can do to get this merged. 

Thank you!

### CODE REVIEW

File: apitable.go [hunk 1]
```diff
@@ -23,7 +23,12 @@ package docx
 // AddTable add a new table to body by col*row
 //
 // unit: twips (1/20 point)
-func (f *Docx) AddTable(row int, col int) *Table {
+func (f *Docx) AddTable(
+	row int, 
+	col int, 
+	tableWidth int64, 
+	borderColors [6]string,
```
Comment (fumiama) [inline apitable.go:30]: You can use a struct with 6 direction fields to make the parameter more human-friendly.

File: apitable.go [hunk 2]
```diff
@@ -162,3 +190,14 @@ func (c *WTableCell) Shade(val, color, fill string) *WTableCell {
 	}
 	return c
 }
+
+//Replaces any index that is blank with "#000000"
+func CheckBorderColors (borderColors [6]string) [6]string {
```
Comment (fumiama) [inline apitable.go:195]: This can be private.

**Review by fumiama — CHANGES_REQUESTED** [review]
Thanks!

File: apitable.go (hunk 2 above)
Comment (dcharles525) [reply apitable.go:191]: Removed all after changing to struct

File: apitable.go [hunk 3]
```diff
@@ -23,7 +23,12 @@ package docx
 // AddTable add a new table to body by col*row
 //
 // unit: twips (1/20 point)
-func (f *Docx) AddTable(row int, col int) *Table {
+func (f *Docx) AddTable(
+	row int, 
+	col int, 
+	tableWidth int64, 
+	borderColors TableBorderColors,
```
Comment (fumiama) [inline apitable.go:30]: The param should be pointer type *TableBorderColors

File: apitable.go (hunk 2 above)
Comment (fumiama) [reply apitable.go:191]: So you can remove this change. But the default value is still `""` so you can add a private method to the struct `TableBorderColors` like
```go
func (tbc *TableBorderColors) applydefault() {
    // look each field and fill "#000000" to "" value
}
```

File: structtable.go [hunk 4]
```diff
@@ -115,6 +115,14 @@ type WTableProperties struct {
 	Look          *WTableLook
 }
 
+type TableBorderColors struct {
```
Comment (fumiama) [inline structtable.go:118]: The name should be `APITableBorderColors` and you can move this definition to where it is used, `apitable.go`.

**Review by fumiama — CHANGES_REQUESTED** [review]

**Review by fumiama — CHANGES_REQUESTED** [review]
Thanks!

**Comment by dcharles525** [comment]
> Thanks!

Absolutely! Thank you for bearing with me, I am new to Go :)

**Comment by fumiama** [comment]
> > Thanks!
> 
> Absolutely! Thank you for bearing with me, I am new to Go :)

It doesn't matter. Everyone start from new and you learn it really fast 👍. I noticed that there's some ident mismatches in your code. Maybe you should use the formatter `gofmt` that go standard suggested. Generally, VSCode will suggest you to install it and format the code automatically when you press `ctrl+s`.

**Pull request merged**